File: benchmarks/phase0/corpus/java/Flows.java
final class Flows {
    void alternateSource() throws Exception {
        String command = System.getenv("COMMAND");
        Runtime.getRuntime().exec(command); // phase0-expect vuln
    }

    void directSource(HttpServletRequest request) throws Exception {
        String command = request.getParameter("command");
        Runtime.getRuntime().exec(command); // phase0-expect vuln
    }

    void constantSink(HttpServletRequest request) throws Exception {
        request.getParameter("ignored");
        Runtime.getRuntime().exec("echo safe"); // phase0-expect safe
    }
}

File: benchmarks/phase0/corpus/go/flows.go
package phase0

import (
	"net/http"
	"os"
	"os/exec"
)

func alternateSource() {
	command := os.Getenv("COMMAND")
	exec.Command("sh", "-c", command) // phase0-expect vuln
}

func directSource(request *http.Request) {
	command := request.FormValue("command")
	exec.Command("sh", "-c", command) // phase0-expect vuln
}

func constantSink(request *http.Request) {
	request.FormValue("ignored")
	exec.Command("echo", "safe") // phase0-expect safe
}
